### PR TITLE
Improvement javaclient 156 add fetch blob as stream api

### PR DESCRIPTION
**Pull request by kevinleturc**

### CODE REVIEW

File: nuxeo-java-client/src/main/java/org/nuxeo/client/MediaType.java [hunk 1]
```diff
@@ -38,7 +38,18 @@
...
+
+    /**
+     * @since 3.1
+     */
+    protected static final String DEFAULT_STRING = "application/octet-stream";
+
+    /**
+     * @since 3.1
+     */
+    protected static final MediaType DEFAULT = parse("application/octet-stream");
```
Comment (covolution) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/MediaType.java]: That could be `parse(DEFAULT_STRING)`

File: nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java [hunk 2]
```diff
@@ -341,15 +344,30 @@ public void onFailure(Call<T> call, Throwable t) {
                     }
                 }
                 if (filename == null) {
-                    filename = fbody.getFilename();
+                    filename = blob.getFilename();
                 }
 
                 String mimeType = headers.get("Content-Type");
```
Comment (covolution) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java]: Constant `HttpHeaders.CONTENT_TYPE` ?

File: nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java [hunk 3]
```diff
@@ -341,15 +344,30 @@ public void onFailure(Call<T> call, Throwable t) {
...
                 if (mimeType == null) {
                     mimeType = MediaTypes.APPLICATION_OCTET_STREAM_S;
                 }
-                FileBlob fileBlob = new FileBlob(fbody.getFile(), filename, mimeType);
-                return retrofit2.Response.success((T) fileBlob, response.raw());
+                String lengthString = headers.get("Content-Length");
```
Comment (covolution) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/NuxeoClient.java]: Constant `HttpHeaders.CONTENT_LENGTH`

File: nuxeo-java-client/src/main/java/org/nuxeo/client/objects/blob/Blob.java [hunk 4]
```diff
@@ -33,6 +33,11 @@
 
     int getLength();
 
+    /**
+     * @since 3.1
+     */
+    long getLongLength();
```
Comment (covolution) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/objects/blob/Blob.java]: hmm, a bit messy to put the type in the method name.  I don't have a better solution, other than changing the return type of `getLength`!  Is `getLength` deprecated?

Comment (covolution) [reply nuxeo-java-client/src/main/java/org/nuxeo/client/objects/blob/Blob.java]: How about `getContentLength`?

Comment (kevinleturc) [reply nuxeo-java-client/src/main/java/org/nuxeo/client/objects/blob/Blob.java]: I should deprecate `getLength` indeed, agreed for `getContentLength`

**Comment by covolution** [comment]
:+1:

**Comment by troger** [comment]
:+1:

**Pull request merged**